MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -1,72 +1,130 @@
+        QuizQuestion(image: "The Dark Knight", text: questionString, correctAnswer: true),
+        QuizQuestion(image: "Kill Bill", text: questionString, correctAnswer: true),
+        QuizQuestion(image: "The Avengers", text: questionString, correctAnswer: true),
+        QuizQuestion(image: "Deadpool", text: questionString, correctAnswer: true),
+        QuizQuestion(image: "The Green Knight", text: questionString, correctAnswer: true),
+        QuizQuestion(image: "Old", text: questionString, correctAnswer: false),
+        QuizQuestion(image: "The Ice Age Adventures of Buck Wild", text: questionString, correctAnswer: false),
+        QuizQuestion(image: "Tesla", text: questionString, correctAnswer: false),
+        QuizQuestion(image: "Vivarium", text: questionString, correctAnswer: false)
+    ]
+    
+    @IBOutlet private var imageView: UIImageView!
+    @IBOutlet private var textLabel: UILabel!
+    @IBOutlet private var counterLabel: UILabel!
+    
     // MARK: - Lifecycle
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        show(quiz: getCurrentQuizStepViewModel())
+    }
+    
+    private func getAppColor(_ name: String) -> CGColor {
+        if let color = UIColor(named: name) {
+            return color.cgColor
+        } else {
+            return UIColor.white.cgColor
+        }
+    }
+    
+    @IBAction private func yesButtonClicked(_ sender: UIButton) {
+        showAnswerResult(isCorrect: questions[currentQuestionIndex].correctAnswer == true)
+    }
+    
+    @IBAction private func noButtonClicked(_ sender: UIButton) {
+        showAnswerResult(isCorrect: questions[currentQuestionIndex].correctAnswer == false)
+    }
+    
+    private func getCurrentQuizStepViewModel() -> QuizStepViewModel {
+        return convert(model: questions[currentQuestionIndex])
+    }
+    
+    private func showAnswerResult(isCorrect: Bool) {
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 8
+        imageView.layer.cornerRadius = 20
+        imageView.layer.borderColor = isCorrect ? getAppColor("ypGreen") : getAppColor("ypRed")
+        
+        if isCorrect {
+            correctAnswers += 1
+        }
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { // запускаем задачу через 1 секунду
+            self.showNextQuestionOrResults()
+        }
+    }
+    
+    private func show(quiz step: QuizStepViewModel) {
+        imageView.image = step.image
+        textLabel.text = step.question
+        counterLabel.text = step.questionNumber
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 0
+        imageView.layer.cornerRadius = 20
+    }
+    
+    private func show(quiz result: QuizResultsViewModel) {
+        let alert = UIAlertController(title: result.title, message: result.text, preferredStyle: .alert)
+        let action = UIAlertAction(title: result.buttonText, style: .default) { [self] _ in
```

Comment:
лучше захватывать со слабой ссылкой, либо пока что вовсе не захватывать :) 
```suggestion
        let action = UIAlertAction(title: result.buttonText, style: .default) { [weak self] _ in
```